src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/PackageWrapper.cs

```diff
@@ -28,14 +32,22 @@ public PackageWrapper(string Name, string FullName, string FamilyName, bool IsFr
 
         public static PackageWrapper GetWrapperFromPackage(Package package)
         {
-            return new PackageWrapper(
+            try
+            {
+                return new PackageWrapper(
                         package.Id.Name,
                         package.Id.FullName,
                         package.Id.FamilyName,
                         package.IsFramework, 
                         package.IsDevelopmentMode,
                         package.InstalledLocation.Path
                         );
+            }
+            catch(FileNotFoundException ex)
+            {
+                ProgramLogger.LogException($"PackageWrapper", "GetWrapperFromPackage","package.InstalledLocation.Path",$"File Not Found for package {package.Id.Name}", ex);
+                return new PackageWrapper();
```

Comment:
I think would it be better to add `try-catch` in places where `GetWrapperFromPackage` is called. This is because now we will be hitting two errors : 
1. When `PackageWrapper` cannot be created and it returns an empty `PackageWrapper`. 
2. When it tries to use an empty `PackageWrapper` and calls `InitializeAppInfo` object.

This would create two redundant logs for errors for the same issue. Also, handling exceptions is costly as mentioned [here](https://stackoverflow.com/questions/2075151/try-catch-blocks-always-expensive), so we should minimize the number of exceptions that we hit. 

Another way to solve this issue would be to use try-catch here, return `null` and do a null check later.

Reply:
Two exceptions will not be thrown, it is just one in the `PackageWrapper` which returns the empty PackageWrapper. In `CurrentUserPackages` the invalid apps are filtered if the path is an empty string (without throwing an exception). Please let me know if you need further clarification. I think this is a neat way so that even if in the future, we add some code which uses `GetWrapperFromPackage` we wouldn't be crashing PT Run if we forget to add a catch block.

Reply:
What about the case for `Storage>PackageRepository>OnPackageInstalling` function ?

Reply:
Yea, an exception would be thrown there. I shall add a validity check like in in CurrentUserPackages to see if the `path` is not null or empty so that we wouldn't have to hit the exception.

Reply:
Sounds good!

Reply:
@somil55, I have a general question, is there a reason why args.Package.InstalledLocation.Path is being used in `OnPackageInstalling`? I think we can reuse the packageWrapper. Thoughts?

Reply:
> @somil55, I have a general question, is there a reason why args.Package.InstalledLocation.Path is being used in `OnPackageInstalling`? I think we can reuse the packageWrapper. Thoughts? That way we wouldn't have to have that catch block.

I think this is a good idea.

Reply:
@alekhyareddy28 Sure! That would improve code readability. But I think you would still need a try/catch block because it's possible that `AppxManifest.xml` file is missing/corrupted in which case an error would be thrown in `InitializeAppInfo`.

Reply:
I think we would still need the catch block if the appmanifest.xml file does not exist for a package. I have left the catch block as is but reused packagewrapper so that we wouldn't throw an exception again while accessing args.InstalledLocation.Path.